File: 1260. Shift 2D Grid/_1260.py
class Solution:
    def shiftGrid(self, grid, k: int):
        res  = list()

        k = k%(len(grid)*len(grid[0]))

        for i in grid:
            for j in i:
                res.append(j)
        
        res = res[-k:] + res[:len(res)-k]

        s = list()
        for i in range(0, len(grid)):
            tmp = list()
            for j in range(0, len(grid[0])):
                tmp.append(res[i*len(grid[0])+j])
            s.append(tmp)

        return s

    def shiftGrid1(self, grid , k: int) :
        f1 = lambda col, nums: [nums[i:i+col] for i in range(0, len(nums), col)]
        f2 = lambda nums, k: nums[-k%len(nums):] + nums[:-k%len(nums)]
        return f1(len(grid[0]), f2(sum(grid, []), k))

File: 1260. Shift 2D Grid/1260.go
package leetcode

import(
	//"fmt"
)

func shiftGrid(grid [][]int, k int) [][]int {
	res := make([]int,0 ,len(grid)*len(grid[0]))
	
	k = k%(len(grid)*len(grid[0]))

	for i:=0; i<len(grid); i++{
		for j:=0; j<len(grid[0]); j++{
			res = append(res, grid[i][j])
		}
	}

	s := make([][]int, len(grid))
    for i := 0; i < len(grid); i++ {
        s[i] = make([]int, len(grid[0]))
	}

	res1 := make([]int, len(grid)*len(grid[0]))
	for i, v := range res[len(res)-k:len(res)]{
		res1[i] = v
	}

	for i,v := range res[:len(res)-k]{
		res1[i+k] = v
	}

	for i:=0; i<len(grid); i++{
		//fmt.Println(s)
		for j:=0; j<len(grid[0]); j++{
			s[i][j] = res1[i*len(grid[0])+j]
		}
	}

	return s
}